synapse/handlers/federation_event.py

```diff
@@ -766,15 +766,22 @@ async def _process_pulled_event(
         """
         logger.info("Processing pulled event %s", event)
 
-        # TODO: Why does this matter? The whole point of this function is to
-        # persist random PDU's from backfill. It shouldn't matter whether we saw
-        # them somewhere else first as an outlier, then during backfill. This
-        # function handles de-outliering anyway.
+        # This function should only be used to backfill events. If you're trying
```

Comment:
> This function should only be used to backfill events

I'm not sure this is either true or particularly helpful. Notably, it rather raises the question of "wtf is the `backfilled` parameter for?". As well as "backfilled" events, it is used to process events returned from `get_missing_events`, which are treated in the much the same way as those *pushed* via a federation `/send`.

I can't really think of a more helpful thing to say "This function should not be used to persist outliers".

Reply:
Updated in https://github.com/matrix-org/synapse/pull/13320